modules/caddyhttp/server.go

```diff
@@ -248,6 +249,14 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		wrec := NewResponseRecorder(w, nil, nil)
 		w = wrec
 
+		// wrap the request body in a LengthReader
+		// so we can track the number of bytes read from it
+		var bodyReader *lengthReader
```

Comment:
Might be a long shot, but does this have to be a pointer? Just wondering if we can avoid an allocation.

Reply:
I don't think so because `r.Body` takes a pointer

> cannot use bodyReader (variable of type lengthReader) as io.ReadCloser value in assignment: lengthReader does not implement io.ReadCloser (method Close has pointer receiver) compiler([InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#InvalidIfaceAssign))

Reply:
Well maybe if I make Read and Close not use a pointer receiver, but does that even work?

Reply:
Ah yeah, that was my concern. Makes sense. Oh well!

Reply:
You were kinda right though - we should probably use `sync.Pool` for `lengthReader`.